zkllvm-blueprint.nix

```diff
@@ -0,0 +1,128 @@
+{ lib,
+  stdenv,
+  src_repo,
+  ninja,
+  pkg-config,
+  cmake,
+  boost183,
+  # We'll use boost183 by default, but you can override it
+  boost_lib ? boost183,
+  gdb,
+  cmake_modules,
+  crypto3,
+  enableDebugging,
+  enableDebug ? false,
+  runTests ? false,
+  }:
+let
+  inherit (lib) optional;
+in stdenv.mkDerivation rec {
+  name = "blueprint";
+
+  src = src_repo;
+
+  nativeBuildInputs = [ cmake ninja pkg-config ] ++ (lib.optional (!stdenv.isDarwin) gdb);
+
+  # enableDebugging will keep debug symbols in boost
+  propagatedBuildInputs = [ (if enableDebug then (enableDebugging boost_lib) else boost_lib) ];
+
+  buildInputs = [cmake_modules crypto3];
+
+  testList = [
```

Comment:
Why do we need this list here? Can we do things on the cmake level?

Reply:
Agree, it would be better to have it on the cmake level.
However, I can define a compilation target for all those tests, but I can't iterate over them to feed it to ctest.
I would keep it like this for now.
Researchers should disable/comment out all the tests outside of the list, then the whole thing can be simpified. But not now, unfotunatelly.

Reply:
Why do it on the level of nix? Why not move it to a shell script that people will be able to run in "nix develop"?
Doing this on the level of nix prevents you from testing interactively.